luma_core/src/sysconf.rs

```diff
@@ -0,0 +1,184 @@
+//! ``sysconf`` module of ``luma_core``.
+//!
+//! Provides access to the Wii system settings, as set by the system menu.
+
+use crate::ios;
+use alloc::collections::BTreeMap;
+use alloc::string::{String, ToString};
+use alloc::vec::Vec;
+use core::fmt;
+
+#[derive(Debug, Clone)]
+enum Item {
+    BigArray(Vec<u8>),
+    SmallArray(Vec<u8>),
+    Byte(u8),
+    Short(u16),
+    Long(u32),
+    LongLong(u64),
+    Bool(bool),
+}
+
+#[derive(Clone, Copy)]
+pub enum AspectRatio {
+    A4_3,
+    A16_9,
+    Unknown,
+}
+
+impl fmt::Debug for AspectRatio {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            AspectRatio::A4_3 => write!(fmt, "4:3"),
+            AspectRatio::A16_9 => write!(fmt, "16:9"),
+            AspectRatio::Unknown => write!(fmt, "unknown"),
+        }
+    }
+}
+
+#[derive(Clone, Copy)]
+pub enum Scan {
+    Interlaced,
+    Progressive,
+    Unknown,
+}
+
+impl fmt::Debug for Scan {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Scan::Interlaced => write!(fmt, "interlaced"),
+            Scan::Progressive => write!(fmt, "progressive"),
+            Scan::Unknown => write!(fmt, "unknown"),
+        }
+    }
+}
+
+#[derive(Clone, Copy)]
+pub enum RefreshRate {
+    R50,
+    R60,
+}
+
+impl fmt::Debug for RefreshRate {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            RefreshRate::R50 => write!(fmt, "50 Hz"),
+            RefreshRate::R60 => write!(fmt, "60 Hz"),
+        }
+    }
+}
+
+/// Parsed structure containing the system settings.
+pub struct Sysconf(BTreeMap<String, Item>);
```

Comment:
I don't think we need something like this. We can make something similar without relying on allocations,